### PR TITLE
feat(DataMapper): Use TypeaheadSelect instead of Select in FieldOverrideModal for type field

### LINKED ISSUE #3344: DataMapper: Improve Field Override candidates dropdown
2 things to improve on Field Override candidates dropdown (both type override and substitution):
* Support typeahead filtering for when there're many candidates, possibly use existing `TypeaheadSelect.tsx` used by AttachSchemaModal root element selection
* Show namespace URI

### How it looks currently
Image

### AttachSchema Modal root element selection (showing namespace URI as a description)
Image

### PR DESCRIPTION
**Pull request by pvillant**
Resolves: [#3344](https://github.com/KaotoIO/kaoto/issues/3344)

https://github.com/user-attachments/assets/782f9b3c-2f72-4678-9c43-beb8f46c91f4

## Summary by CodeRabbit

* **New Features**
  * Upgraded field override type picking to searchable typeahead with improved namespace-based option display.
  * Added support for customizing typeahead dropdown positioning, and improved option rendering when description data is missing.
* **Bug Fixes**
  * Improved typeahead interaction (blur/typing behavior) and prevented unintended `onChange` during open/empty/clear flows.
  * Updated override save behavior to only save type selections when the required type info is available.
  * Schema upload now skips re-reading already-uploaded files and ignores empty selections.
* **Tests**
  * Expanded coverage for typeahead interactions, option formatting, override saving edge cases, and schema upload behavior.

### CODE REVIEW

**Comment by pvillant** [comment]
@coderabbitai review

**Comment by pvillant** [comment]
@coderabbitai review

**Review by mmelko — DISMISSED** [review]
LGTM 👍

File: packages/ui/src/components/Document/actions/FieldOverride/override-util.ts [hunk 1]
```diff
@@ -76,11 +82,30 @@ export function getOverrideCandidates(
...
+  const candidates: Record<string, CandidateDisplay> = {};
+
+  for (const [key, value] of Object.entries(rawCandidates)) {
+    const namespaceURI = isSubstitution
+      ? (value as IFieldSubstituteInfo).qname.getNamespaceURI()
+      : (value as IFieldTypeInfo).typeQName.getNamespaceURI();
+    candidates[key] = {
+      displayName: value.displayName,
+      description: value.description,
+      namespaceURI: namespaceURI ?? 'N/A',
```
Comment (igarashitm) [inline packages/ui/src/components/Document/actions/FieldOverride/override-util.ts]: We don't need the tailing `?? 'N/A'` as `namespaceURI` is always string here.
When the one is in blank namespace, namespace URI will be just an empty string and we can skip showing namespace URI in this case.
In smaller and not widely shared XML schema, they tend to use blank namespace since it doesn't have a risk for name collision, and in that case, user doesn't care about the namespace at all.

File: packages/ui/src/components/Document/actions/FieldOverride/FieldOverrideModal.tsx [hunk 2]
```diff
@@ -173,34 +163,31 @@ export const FieldOverrideModal: FunctionComponent<FieldOverrideModalProps> = ({
...
-    () => Object.entries(candidates).sort(([, a], [, b]) => a.displayName.localeCompare(b.displayName)),
-    [candidates],
-  );
+  const typeaheadOptions = useMemo<TypeaheadSelectOption[]>(() => {
+    return Object.entries(candidates)
+      .sort(([, a], [, b]) => a.displayName.localeCompare(b.displayName))
+      .map(([key, candidate]) => {
+        const parts: string[] = [];
+        if (candidate.namespaceURI) parts.push(`Namespace URI: ${candidate.namespaceURI}`);
+        if (candidate.description) parts.push(candidate.description);
```
Comment (igarashitm) [inline packages/ui/src/components/Document/actions/FieldOverride/FieldOverrideModal.tsx]: I would only show namespace URI here - yes, I understand showing description might help for some users, but it's a bit noisy for who knows about the schema well and just want to see the information to distinguish same name but different namespace ones.

**Review by igarashitm — COMMENTED** [review]
A few minor comments, LGTM otherwise 👍

**Comment by pvillant** [comment]
Screenshot 2026-07-07 at 10 44 53

**Comment by pvillant** [comment]
@coderabbitai review

**Review by igarashitm — APPROVED** [review]
